File: readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int count_letters(string s);
int count_words(string s);
int count_sentences(string s);

int main(void)
{
    string t = get_string("Text: ");
    
    //printf("%i letter(s)\n", count_letters(t));
    //printf("%i word(s)\n", count_words(t));
    //printf("%i sentence(s)\n", count_sentences(t));
    
    float letters = count_letters(t);
    float words = count_words(t);
    float sentences = count_sentences(t);
    
    float l = (letters / words)*100;
    float s = (sentences / words)*100;
    
    float index = 0.0588 * l - 0.296 * s - 15.8;
    
   // printf("%f, %f, %f\n",l, s, index);
    
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %.f\n", round(index));
    }
        
}

int count_letters(string s)
{
    int sum = 0;
    for (int i = 0, n = strlen(s); i < n; i++)
    {
        if (isalpha(s[i]))
        {
            sum += 1;
        }
    }
    
    return sum;
}

int count_words(string s)
{
    int sum = 1;
    for (int i = 0, n = strlen(s); i < n; i++)
    {
        if (isspace(s[i]))
        {
            sum += 1;
        }
    }
    return sum;
}

int count_sentences(string s)
{
    int sum = 0;
    for (int i = 0, n = strlen(s); i < n; i++)
    {
        if ((s[i] == '?' ) || (s[i] == '.') || (s[i] == '!'))
        {
            sum += 1;
        }
    }
   return sum;
    
}

File: readability.py
import re


def main():

    text = input("Text: ")
    
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)
    
    l = (letters / words) * 100
    s = (sentences / words) * 100
    
    count = 0.0588 * l - 0.296 * s - 15.8
    
    if count < 1:
        print("Before Grade 1")
    elif count > 16:
        print("Grade 16+")
    else:
        print("Grade {}".format(round(count)))
        

def count_letters(s):
    
    return len(re.findall(r'\w', s))
    
    # sum = 0
    # for i in range(len(s)):
    #     if s[i].isalpha():
    #         sum += 1
    # return sum
    

def count_words(s):
    return len(re.findall(r'\s', s))+1
    
    # sum = 1
    # for i in range(len(s)):
    #     if s[i].isspace():
    #         sum += 1
    # return sum

    
def count_sentences(s):
    return len(re.findall(r'[.|?|!]', s))
    
    # sum = 0
    # for i in range(len(s)):
    #     if (s[i] == '?') or (s[i] == '.') or (s[i] == '!'):
    #         sum += 1
    # return sum
    

main()